### PR TITLE
Update reference to WPAuthenticator

### PR DESCRIPTION
**Pull request by ctarda**
Fxes #15394 

This is a maintenance PR to update the dependency with WPAuthenticator.

The new beta release of WPAuthenticator includes a minor breaking change to the API of WordPressAuthenticatorDelegate.

## Changes
* Bump WPAuthenticator to 1.30.0-beta.2
* Update WordPressAuthenticatorManager to implement the changes required by WPAuthenticator

## To test
* Checkout the branch. Run `bundle exec pod install`
* Log out if necessary
* Log in using both "Enter your site address"  and "Continue with WordPress.com"

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -198,7 +198,7 @@ target 'WordPress' do
 
     pod 'Gridicons', '~> 1.0.2'
 
-    pod 'WordPressAuthenticator', '~> 1.29.0'
+    pod 'WordPressAuthenticator', '~> 1.30.0-beta.2'
```
Comment (ScoutHarris) [inline Podfile]: We don't include the beta number, so it should just be `1.30.0-beta`.

**Review by ScoutHarris — APPROVED** [review]
After the version is updated, :shipit: . Thanks for doing this!

File: Podfile (hunk 1 above)
Comment (ctarda) [reply Podfile]: Oh, I missed that, thanks!

**Comment by ctarda** [comment]
Version is updated and CI is ✅ . Merging!

**Pull request merged**